apps/datadogcharts/datadog_charts.star

```diff
@@ -0,0 +1,160 @@
+""")
+
+def main(config):
+    # Setup and validate config
+    DD_API_KEY = config.get("api_key") or DEFAULT_API_KEY
+    DD_APP_KEY = config.get("app_key") or DEFAULT_APP_KEY
+    DASHBOARD_ID = config.get("dashboard_id")
+
+    if DD_API_KEY == None or DD_APP_KEY == None:
+        child = render.Row(
+            cross_align = "center",
+            main_align = "center",
+            children = [
+                render.WrappedText(align = "center", content = "DataDog API Key or App Key not set."),
+            ],
+        )
+        return render.Root(child = child)
+
+    dashboard_json = http.get(
```

Comment:
Not sure if it's needed or not, but note that you can pass `ttl_seconds=4711` to cache the response. In case datadog prefers less traffic. The HTTP module should also respect cache headers by default, so that may be sufficient.

Reply:
That's a lot easier of a caching solution that I was playing with before! I'll add it.